app/goal_routes.py

```diff
@@ -0,0 +1,128 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+from app.task_routes import validate_task
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_goal(id):
```

Comment:
Since validate_goal is almost identical to validate_task, consider pulling it out into a helper file and importing the helper file in goal_routes.py and task_routes.py to avoid repeating code.